### PR TITLE
 SNAP-892 client id validation

### PR DESCRIPTION
**Pull request by pokharelmanish**
### Jira Story

- [client-id validation](https://osi-cwds.atlassian.net/browse/SNAP-892)

## Description
Added validation for client id.

screen shot 2019-03-05 at 10 22 22 am

screen shot 2019-03-05 at 10 22 31 am

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have ran all tests for the project.
- [ ] I have ran lint/rubocop check for the changed/new files.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by pokharelmanish** [comment]
retest this please

File: spec/javascripts/components/common/search/PersonSearchNameGroupSpec.jsx [hunk 1]
```diff
@@ -46,6 +47,29 @@ describe('PersonSearchNameGroup', () => {
...
 
+  describe('clientIdError', () => {
+    it('display error message if clientIdError is present', () => {
+      const component = render({
+        clientIdError: ['Client Id number must be 19 digits long.'],
+      }).find('MaskedInputField[label="Client ID"]')
+      expect(component.props().errors).toEqual(['Client Id number must be 19 digits long.'])
+    })
+
+    it('doesnot display error message if clientIdError is not present', () => {
```
Comment (ggomez-cambriasolutions) [inline spec/javascripts/components/common/search/PersonSearchNameGroupSpec.jsx]: does not

File: spec/javascripts/components/common/search/PersonSearchNameGroupSpec.jsx [hunk 2]
```diff
@@ -46,6 +47,29 @@ describe('PersonSearchNameGroup', () => {
...
+      const component = render({
+        clientIdError: [],
+      }).find('MaskedInputField[label="Client ID"]')
+      expect(component.props().errors).toEqual([])
+    })
+
+    it('doesnot display error message if clientIdError is undefined ', () => {
```
Comment (ggomez-cambriasolutions) [inline spec/javascripts/components/common/search/PersonSearchNameGroupSpec.jsx]: does not

File: spec/javascripts/selectors/peopleSearchSelectorsSpec.js [hunk 3]
```diff
@@ -830,4 +831,27 @@ describe('peopleSearchSelectors', () => {
       expect(selectAkaFullName(state, result)).toEqual(null)
     })
   })
+
+  describe('selectClientIdError', () => {
+    it('doesnot returns error message if searchClientId is 19 digits.', () => {
```
Comment (ggomez-cambriasolutions) [inline spec/javascripts/selectors/peopleSearchSelectorsSpec.js]: does not return...

File: spec/javascripts/selectors/peopleSearchSelectorsSpec.js [hunk 4]
```diff
@@ -830,4 +831,27 @@ describe('peopleSearchSelectors', () => {
...
+      const peopleSearch = {searchClientId: '1111-1111-1111-1111111'}
+      const state = fromJS({peopleSearch})
+      expect(selectClientIdError(state))
+        .toEqual([])
+    })
+
+    it('returns error message if it searchClientId is 12 digits.', () => {
```
Comment (ggomez-cambriasolutions) [inline spec/javascripts/selectors/peopleSearchSelectorsSpec.js]: ...message if searchClientId is...

File: spec/javascripts/selectors/peopleSearchSelectorsSpec.js [hunk 5]
```diff
@@ -830,4 +831,27 @@ describe('peopleSearchSelectors', () => {
...
+      const peopleSearch = {searchClientId: '1111-1111-1111-_______'}
+      const state = fromJS({peopleSearch})
+      expect(selectClientIdError(state))
+        .toEqual(['Client Id number must be 19 digits long.'])
+    })
+
+    it('returns error message if it searchClientId less than 19 digits.', () => {
```
Comment (ggomez-cambriasolutions) [inline spec/javascripts/selectors/peopleSearchSelectorsSpec.js]: ...message if searchClientId is...

File: spec/javascripts/utils/clientIdValidatorSpec.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import {getClientIdErrors} from 'utils/clientIdValidator'
+
+describe('clientIdValidator', () => {
+  describe('getClientIdErrors', () => {
+    it('doesnot returns error message if client id is 19 digits.', () => {
```
Comment (ggomez-cambriasolutions) [inline spec/javascripts/utils/clientIdValidatorSpec.js]: does not return error message if client id is...

**Comment by pokharelmanish** [comment]
@ggomez-cambriasolutions Ready to review again.

**Pull request merged**